index.html

```diff
@@ -3070,25 +3087,121 @@ <h2>
             <dd>
 If the resolution is successful, this MUST be a <a
 href="metadata-structure">metadata structure</a>. This structure contains
-metadata about the <a>DID document</a> contained in the
+metadata about the <a>DID document</a> contained in the <code>did-document</code> or
 <code>did-document-stream</code>. This metadata typically does not change
 between invocations of the <code>resolve</code> function unless the <a>DID
 document</a> changes, as it represents data about the <a>DID document</a>. If
 the resolution is unsuccessful, this output MUST be an empty <a
 href="metadata-structure">metadata structure</a>.
+Properties defined by this specification are in <a href="#did-document-metadata-properties"></a>.
             </dd>
         </dl>
 
         <p>
-<a>DID resolver</a> implementations MUST NOT alter the signature of this
-function in any way. <a>DID resolver</a> implementations MAY map the
-<code>resolve</code> function to a method-specific internal function to perform
+<a>DID resolver</a> implementations MUST NOT alter the signature of these
+functions in any way. <a>DID resolver</a> implementations MAY map the
```

Comment:
> <a>DID resolver</a> implementations MUST NOT alter the signature of these
functions in any way.

I understand the intent here but I imagine it may trip up implementers ... at least until we get more specific by using, for example, Infra. Then it may be ok to say this. The fact that we already have a normative statement saying that resolvers must implement this function could also be seen as enough and could avoid issues here.

Reply:
I think this sentence originally came from @jricher . I'd be fine with removing.

Reply:
+1 to just remove it -- since I was also thinking about how there may be some implementations that could provide additional optional parameters, for example, that would not run afoul of any interoperability concerns here. We don't need to be this prescriptive.

Reply:
The goal is to ensure interop. If you have additional functions (which might be how you handle optional parameters) then that's fine.